### DISCUSSION TITLE
Anybody tried LLMLingua ?

### DISCUSSION THREAD

**Discussion by x4080** [ggml-org/llama.cpp#4711] [General]
https://github.com/microsoft/LLMLingua/tree/main

It tried to compress prompt and document so can be much smaller, I just wondered if we can use it with llama cpp (gguf)

**Comment by Michael-F-Ellis** [comment]
Seems really interesting.  Microsoft developed it as a way to save $ when interacting with paid servers like OpenAI, but I'm wondering if it could be beneficial for llama.cpp. MS claim the method can achieve prompt compressions of up to 20x and get  ~same response from the LLM. I'd certainly love to have 160K effective context for a mistral model!

**Comment by mirek190** [comment]
Can you imagine?  Current 32K context models upgraded to 640K ..wow.
